### PR TITLE
Implement AUTH

### PR DESCRIPTION
**Pull request by UlugbekAsadov**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+const secretKey = 'sizning_xavfsizlik_kalitingiz';
+
+const payload = {
+  username: 'foydalanuvchi',
+  exp: Math.floor(Date.now() / 1000) + (60 * 60) // Token amal qilish muddati: 1 soat
+};
+
+const token = jwt.sign(payload, secretKey);
+console.log(token);
```
Comment (UlugbekAsadov) [inline app.js]: remove console.log()

File: app.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+// import { Vonage } from "@vonage/server-sdk"
```
Comment (UlugbekAsadov) [inline app.js]: remove commented code

File: src/config/key.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const ACCESS_TOKEN = "349f1c32fac5bb1a50583d87f5f24b25sdf6ds786fsd8yf8sdfl32873udj32093u4jer8w983n4eirsv34v23jk4bil23g4vl32bg4"
```
Comment (UlugbekAsadov) [inline src/config/key.js]: .env file ga o'tkazish kerak

`JWT_ACCESS_TOKEN=349f1c32fac5bb1a50583d87f5f24b25sdf6ds786fsd8yf8sdfl32873udj32093u4jer8w983n4eirsv34v23jk4bil23g4vl32bg4`

File: src/models/user.model.js [hunk 4]
```diff
@@ -10,4 +12,14 @@ const userModel = new Schema({
   phoneNumber: { type: Number, required: true, unique: true },
 });
 
+function generateAuthToken(){
+  const tokenkey = ACCESS_TOKEN
+  const token = jwt.sign({_id: this._id},tokenkey)
```
Comment (UlugbekAsadov) [inline src/models/user.model.js]: remove _id, 
add
role: this.role,
status: this.status

File: src/controllers/auth.controller.js [hunk 5]
```diff
@@ -58,10 +58,9 @@ export const signIn = async (req, res) => {
         role,
         phoneNumber,
       };
-
-      return res.status(200).json({ data: user });
+      const token = await user.generateAuthToken()
```
Comment (UlugbekAsadov) [inline src/controllers/auth.controller.js]: shu narsani sign up qilayotganda ham yuborish kerak

**Review by UlugbekAsadov — COMMENTED** [review]
Nechi kunda token muddati tugaydi?

File: app.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+//     await vonage.sms.send({to, from, text})
+//         .then(resp => { console.log('Message sent successfully'); console.log(resp); })
+//         .catch(err => { console.log('There was an error sending the messages.'); console.error(err); });
+// }
+
+// sendSMS();
+
+import jwt from "jsonwebtoken"
+
+const secretKey = 'sizning_xavfsizlik_kalitingiz';
```
Comment (UlugbekAsadov) [inline app.js]: buni ham .env ga o'tkazish kerak

File: app.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+const payload = {
+  username: 'foydalanuvchi',
+  exp: Math.floor(Date.now() / 1000) + (60 * 60) // Token amal qilish muddati: 1 soat
```
Comment (UlugbekAsadov) [inline app.js]: exp ni o'rniga expiresIn ni ishlating

`expiresIn: process.env.JWT_TOKEN_EXPIRE_DURATION // 1 kun bo'ladi`

.env file
`JWT_TOKEN_EXPIRE_DURATION: 1d`

**Comment by Obuljon** [comment]
const token = jwt.sign({_id: this._id},tokenkey,{expiresIn:"1h"} ) bunda 1
soat
yana qoshish mumkin vaqt talabga qarab


пн, 19 июн. 2023 г. в 15:02, Ulugbek Asadov ***@***.***>:

File: src/utils/tryCatch.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export const tryCatch = (controller) => async (req, res, next) => {
```
Comment (UlugbekAsadov) [inline src/utils/tryCatch.js]: bu funksiyaning o'zining nomi bor `asyncHandler`. `tryCatch` ni `asyncHandler` ga o'zgartiring.

express async handler degan kutubxona ham bor.

tryCatch => asyncHandler

**Comment by Obuljon** [comment]
bu code larni sinap ko'rish uchun yozgandim


пн, 19 июн. 2023 г. в 15:02, Ulugbek Asadov ***@***.***>:

**Pull request merged**